indigo_app/views/tasks.py

```diff
@@ -192,32 +185,21 @@ class TaskChangeStateView(TaskViewBase, View, SingleObjectMixin):
     def post(self, request, *args, **kwargs):
         task = self.get_object()
         user = self.request.user
-
-        if self.change == 'submit':
-            if not has_transition_perm(task.submit, self):
-                raise PermissionDenied
-            task.submit(user)
-            messages.success(request, u"Task '%s' has been submitted for review" % task.title)
-        if self.change == 'cancel':
-            if not has_transition_perm(task.cancel, self):
-                raise PermissionDenied
-            task.cancel(user)
-            messages.success(request, u"Task '%s' has been cancelled" % task.title)
-        if self.change == 'reopen':
-            if not has_transition_perm(task.reopen, self):
-                raise PermissionDenied
-            task.reopen(user)
-            messages.success(request, u"Task '%s' has been reopened" % task.title)
-        if self.change == 'unsubmit':
-            if not has_transition_perm(task.unsubmit, self):
-                raise PermissionDenied
-            task.unsubmit(user)
-            messages.success(request, u"Task '%s' has been unsubmitted" % task.title)
-        if self.change == 'close':
-            if not has_transition_perm(task.close, self):
-                raise PermissionDenied
-            task.close(user)
-            messages.success(request, u"Task '%s' has been closed" % task.title)
+        potential_changes = {
+            'submit': 'submitted',
+            'cancel': 'cancelled',
+            'reopen': 'reopened',
+            'unsubmit': 'unsubmitted',
+            'close': 'closed',
+        }
+
+        for change_made, verb in potential_changes.items():
+            if self.change == change_made:
+                if not has_transition_perm(task.__getattribute__(change_made), self):
```

Comment:
Nice :)

`getattr` is preferred to `task.__getattribute__`, calling underscore methods is generally discouraged. I'd also store the result of that in a variable since you call it twice.